src/main/java/fi/metatavu/kuntaapi/server/controllers/TaskController.java

```diff
@@ -0,0 +1,291 @@
+   * @param taskQueue queue name
+   * @param responsibleNode node name
+   * @return updated task queue
+   */
+  public TaskQueue updateTaskQueueResponsibleNode(TaskQueue taskQueue, String responsibleNode) {
+    return taskQueueDAO.updateResponsibleNode(taskQueue, responsibleNode);
+  }
+
+  /**
+   * Return whether node is responsible from queue
+   * 
+   * @param queueName queue name
+   * @param responsibleNode node name
+   * @return true if node is responsible from queue otherwise false
+   */
+  public boolean isNodeResponsibleFromQueue(String queueName, String responsibleNode) {
+    TaskQueue taskQueue = taskQueueDAO.findByNameAndResponsibleNode(queueName, responsibleNode);
+    return taskQueue != null;
+  }
+  
+  /**
+   * Return whether queue is empty
+   * 
+   * @param queueName queue name
+   * @return true if queue is empty otherwise false
+   */
+  public boolean isQueueEmpty(String queueName) {
+    TaskQueue taskQueue = taskQueueDAO.findByName(queueName);
+    
+    if (taskQueue == null) {
+      return true;
+    }
+    
+    return taskDAO.countByQueue(taskQueue) == 0;
+  }
+  
+  /**
+   * Return whether queue exists
+   * 
+   * @param queueName queue name
+   * @return true if queue exists otherwise false
+   */
+  public boolean isQueueExisting(String queueName) {
+    TaskQueue taskQueue = taskQueueDAO.findByName(queueName);
+    return taskQueue != null;
+  }
+  
+  @SuppressWarnings ("squid:S1168")
+  private <T extends AbstractTask> byte[] serialize(T task) {
+    try (ByteArrayOutputStream resultStream = new ByteArrayOutputStream()) {
+      serializeToStream(task, resultStream);
+      resultStream.flush();
+      return resultStream.toByteArray();
+    } catch (IOException e) {
+      if (logger.isLoggable(Level.SEVERE)) {
+        logger.log(Level.SEVERE, "Failed to write serialized task data", e);
+      }
+    }
+    
+    return null;
+  }
+
+  private <T extends AbstractTask> void serializeToStream(T task, ByteArrayOutputStream resultStream) {
+    try (ObjectOutputStream objectStream = new ObjectOutputStream(resultStream)) {
+      objectStream.writeObject(task);
+      objectStream.flush();
+    } catch (IOException e) {
+      if (logger.isLoggable(Level.SEVERE)) {
+        logger.log(Level.SEVERE, "Failed to serialize task", e);
+      }
+    }
+  }
+
+  private <T extends AbstractTask> T unserialize(byte[] rawData) {
+    try (ByteArrayInputStream byteStream = new ByteArrayInputStream(rawData)) {
+      return unserializeFromStream(byteStream);
+    } catch (IOException e) {
+      if (logger.isLoggable(Level.SEVERE)) {
+        logger.log(Level.SEVERE, "Failed to write unserialized task data", e);
+      }
+    }
+    
+    return null;
+  }
+
+  @SuppressWarnings("unchecked")
+  private <T extends AbstractTask> T unserializeFromStream(ByteArrayInputStream byteStream) {
+    try (ObjectInputStream objectStream = new ObjectInputStream(byteStream)) {
+      Object object = objectStream.readObject();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make sure deserializing objects is safe here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4508)